ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -143,7 +143,13 @@ public Long getValue() {
         return index.hardDeleteCaughtUp() ? 1L : 0L;
       }
     };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "-hardDeleteCaughtUp"),
-        hardDeleteCaughtUp);
+    registry.register(MetricRegistry.name(PersistentIndex.class, name + ".HardDeleteCaughtUp"), hardDeleteCaughtUp);
+  }
+
+  static String getPartitionName(String replicaDir) {
```

Comment:
Yes, I got what you are trying. I am just wondering why partition name is not available in a more simpler way.
It wouldn't be right to assume what the partition id is by processing the path this way. This may not be the format we follow tomorrow, and this may not be correct for all OSes etc.
The StoreManager has access to the ReplicaId. Perhaps you can get the partition id string from it and pass it to the `Store` constructor. Ideally the `StoreMetrics` should not have any "business logic".